resource_obmcs_core_private_ip_test.go

```diff
@@ -17,134 +18,100 @@ type ResourcePrivateIPTestSuite struct {
 	Client       *baremetal.Client
 	Provider     terraform.ResourceProvider
 	Providers    map[string]terraform.ResourceProvider
-	TimeCreated  baremetal.Time
 	Config       string
 	ResourceName string
-	Res          *baremetal.PrivateIP
-	DeletedRes   *baremetal.PrivateIP
 }
 
 func (s *ResourcePrivateIPTestSuite) SetupTest() {
 	s.Client = testAccClient
 	s.Provider = testAccProvider
 	s.Providers = testAccProviders
-	s.TimeCreated = baremetal.Time{Time: time.Now()}
-
-	s.Config = vnicConfig + `
-resource "oci_core_private_ip" "testPrivateIP" {
-	vnic_id = "${lookup(data.oci_core_vnic_attachments.vnics.vnic_attachments[0],"vnic_id")}"
-	display_name = "display_name"
-}
-	`
-
-	s.Config += testProviderConfig()
-
-	s.ResourceName = "oci_core_private_ip.testPrivateIP"
-
+	s.Config = testProvider1() + testADs() + testVCN1() + testSubnet1() + testImage1() + testInstance1() + `
+	data "oci_core_vnic_attachments" "t" {
+		availability_domain = "${data.oci_identity_availability_domains.t.availability_domains.0.name}"
+		compartment_id = "${var.compartment_ocid}"
+		instance_id = "${oci_core_instance.t.id}"
+	}`
+
+	s.ResourceName = "oci_core_private_ip.t"
 }
 
-func (s *ResourcePrivateIPTestSuite) TestCreateResourcePrivateIP() {
-
-	resource.UnitTest(s.T(), resource.TestCase{
+func (s *ResourcePrivateIPTestSuite) TestAccCoreResourcePrivateIP_basic() {
+	var resId string
+	resource.Test(s.T(), resource.TestCase{
 		Providers: s.Providers,
 		Steps: []resource.TestStep{
+			// test create
 			{
 				ImportState:       true,
 				ImportStateVerify: true,
-				Config:            s.Config,
+				Config: s.Config + `
+				resource "oci_core_private_ip" "t" {
+					vnic_id = "${lookup(data.oci_core_vnic_attachments.t.vnic_attachments[0], "vnic_id")}"
+					display_name = "-private-ip"
+				}`,
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(s.ResourceName, "display_name", "display_name"),
 					resource.TestCheckResourceAttrSet(s.ResourceName, "id"),
+					resource.TestCheckResourceAttrSet(s.ResourceName, "vnic_id"),
+					resource.TestCheckResourceAttrSet(s.ResourceName, "ip_address"),
+					resource.TestCheckResourceAttrSet(s.ResourceName, "availability_domain"),
+					resource.TestCheckResourceAttrSet(s.ResourceName, "compartment_id"),
+					resource.TestCheckResourceAttrSet(s.ResourceName, "is_primary"),
+					resource.TestCheckResourceAttrSet(s.ResourceName, "subnet_id"),
+					resource.TestCheckResourceAttrSet(s.ResourceName, "time_created"),
+					resource.TestCheckResourceAttr(s.ResourceName, "display_name", "-private-ip"),
+					func(s *terraform.State) (err error) {
+						resId, err = fromInstanceState(s, "oci_core_private_ip.t", "id")
```

Comment:
Looks like resId is checked for the final destructive change, but might as well verify for the updates as well. No blocking.